api-report/datastore-definitions.api.md

```diff
@@ -37,6 +37,7 @@ export interface IChannel extends IFluidLoadable {
     // (undocumented)
     readonly owner?: string;
     summarize(fullTree?: boolean, trackState?: boolean): ISummaryTreeWithStats;
+    summarizeAsync?(fullTree?: boolean, trackState?: boolean): Promise<ISummaryTreeWithStats>;
```

Comment:
I'm not sure if this is worth attaching in this PR, but we should figure out a way to have consistent naming across all layers. Today, "summarize" means async at IFluidDataStoreChannel layer. We should either have summarize & summarizeAsync, or summarizeSync & summarize (or maybe summarizeAsync & summarizeSync) for sync & async calls respectively.

Reply:
Actually, looking below, can we go with async summarize() and synchronous getAttachSummary()? It will be consistent with data store naming.

Reply:
That said, it maybe harder to do renames (today's summarize() is sync at this layer) due to back-compat requirements. Maybe not - one way or another your PR would need to have back-compat handling.